### DISCUSSION TITLE
Can we know if a managed entity is persisted or not?

### DISCUSSION THREAD

**Discussion by pantajoe** [mikro-orm/mikro-orm#4082] [Q&A] [answered]
Consider some service that receives an entity and depending on whether this entity is a new record created by `em.create` (and persisted by `em.persist`), but it is not yet written inside the database by flushing or just an entity found by `em.find`/`em.findOne`, it performs some action.

Pseudocode example:

```ts
class Service {
  static run(entity: Entity) {
    if (entity.persisted) { // how do we know if the entity is not yet persisted
      // do something
    } else {
      // entity is already persisted in the database
      // do something different
    }
  }
}
```

Is there a way to distinguish between managed entities that are not persisted into the database (prior to flushing) and entities that are inside the database (found by `em.find`)?

**Comment by B4nan** [comment, marked as answer]
A managed entity will be part of the identity map, persisted but not yet flushed entity will be part of persist stack. There is an internal `__managed` flag which should help you find the state directly on the entity instance:

```ts
console.log(wrap(entity, true).__managed);
```

This gets set when we save entity snapshot (which is what the ORM uses for computing the difference on flush).

**Reply by pantajoe** [reply]
Thanks for the quick answer!

So, `wrap(entity, true).__managed` is `true` when the entity is part of the identity map, i.e., persisted in the database;

and it returns `false` when it's a new entity, correct?
Or is it the other way around?

**Reply by B4nan** [reply]
Yes, managed means part of identity map.

**Reply by pantajoe** [reply]
Okay, great. Thanks!
